dotcom-rendering/playwright/tests/parallel-5/atom.video.e2e.spec.ts

```diff
@@ -0,0 +1,458 @@
+	page,
+	videoId,
+	adUnit,
+	pageUrl,
+	rejectAll,
+}: {
+	page: Page;
+	videoId: string;
+	adUnit: string;
+	pageUrl: string;
+	rejectAll: boolean;
+}) => {
+	return page.waitForRequest((request) => {
+		const matchUrl = request
+			.url()
+			.startsWith(`https://www.youtube.com/embed/${videoId}?`);
+		if (!matchUrl) {
+			return false;
+		}
+		const searchParams = new URLSearchParams(request.url());
+		const rawEmbedConfig = searchParams.get('embed_config');
+		if (!rawEmbedConfig) {
+			return false;
+		}
+		const embedConfig = JSON.parse(rawEmbedConfig) as YouTubeEmbedConfig;
+		const adsConfig = embedConfig.adsConfig;
+		const adTagParameters = adsConfig.adTagParameters;
+		// cust_params is double encoded
+		const custParams = new URLSearchParams(
+			decodeURIComponent(adTagParameters.cust_params),
+		);
+		// check adunit
+		const adUnitMatch = adTagParameters.iu === adUnit;
+		// check url to check custParams is present
+		const urlCustParamMatch = custParams.get('url') === pageUrl;
+
+		// check consent related properties
+		// cmpGdpr = consentState.tcfv2.gdprApplies
+		const cmpGdprMatch = adTagParameters.cmpGdpr === 1;
+		// cmpVcd = consentState.tcfv2.tcString
+		const cmpVcdMatch = !isUndefined(adTagParameters.cmpVcd);
+		let nonPersonalizedAdMatch = false;
+		let cmpGvcdMatch = false;
+		if (rejectAll) {
+			// user has not given consent for any purpose
+			nonPersonalizedAdMatch = adsConfig.nonPersonalizedAd;
+			// cmpGvcd = consentState.tcfv2.addtlConsent
+			cmpGvcdMatch = adTagParameters.cmpGvcd === '1~';
+		} else {
+			// user has not given consent for any purpose
+			nonPersonalizedAdMatch = !adsConfig.nonPersonalizedAd;
+			// cmpGvcd = consentState.tcfv2.addtlConsent
+			cmpGvcdMatch = !isUndefined(adTagParameters.cmpGvcd);
+		}
+		return (
+			adUnitMatch &&
+			urlCustParamMatch &&
+			cmpGdprMatch &&
+			cmpVcdMatch &&
+			nonPersonalizedAdMatch &&
+			cmpGvcdMatch
+		);
+	});
+};
+
+/**
+ * Mutes the YouTube player at the given iframe
+ *
+ * Muting one player will mute all other players so it is sufficient to
+ * mute one video on a page with multiple videos.
+ */
+const muteYouTube = async (page: Page, iframeSelector: string) => {
+	const muteButton = page
+		.frameLocator(iframeSelector)
+		.locator('.ytp-mute-button');
+	const muteButtonTitle = await muteButton.getAttribute(
+		'data-title-no-tooltip',
+	);
+	// the mute button is a toggle so check if muting is possible
+	if (muteButtonTitle?.startsWith('Mute')) {
+		await muteButton.click();
+	}
+};
+
+test.describe('YouTube Atom', () => {
+	test('plays main media video', async ({ page }) => {
+		await loadPage(
+			page,
+			'/Article/https://www.theguardian.com/uk-news/2020/dec/04/edinburgh-hit-by-thundersnow-as-sonic-boom-wakes-residents',
+		);
+		await cmpAcceptAll(page);
+
+		await waitForIsland(page, 'YoutubeBlockComponent');
+
+		// Make sure overlay is displayed
+		const overlaySelector = `[data-testid^="youtube-overlay-S0CE1n-R3OY"]`;
```

Comment:
There's a few places where we use this magic string for the video ID. Is it worth extracting to a variable?